### PR TITLE
Add a v1 connector for Saavn

### PR DESCRIPTION
**Pull request by dufferzafar**
I pulled in the original source from [here](http://github.com/mayank127/web-scrobbler/), made some changes and tested it out.

But I can't get the connector to load.

Can someone please have a look and tell what's wrong?

### CODE REVIEW

**Comment by alexesprit** [comment]
Is there a reason to write the connector in v1 format?

**Comment by alexesprit** [comment]
> But I can't get the connector to load.
It's because of wrong URL pattern. It shold be '*://*.saavn.com/*'.

**Comment by dufferzafar** [comment]
> Is there a reason to write the connector in v1 format?

Not really, no. The source I copied it from had it in v1. Once I get this to work, I'll port it to v2, I guess.

**Comment by alexesprit** [comment]
Strange, I see no artist info in the player bar on the Saavn page, like in gaana.com.

File: connectors/saavn.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    return 60 * parseInt(duration[0]) + parseInt(duration[1]);
+}
+
+function updateNowPlaying() {
+
+    var track = getTrack();
+    var duration = getDuration();
+
+    // Extract Track's URL to find artist information
+    var trackURL = $(DOM_TRACK_NAME + ' a').attr("onclick");
```
Comment (dufferzafar) [inline connectors/saavn.js]: @alexesprit  Yes, there is no artist info in the now playing bar, but there is a link to track details, from which we can extract the info.

Comment (alexesprit) [inline connectors/saavn.js]: @dufferzafar ok, I got it. I think we could use the similar mechanism for gaana.com too.

File: connectors/saavn.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                });
+            }
+        });
+    });
+}
+
+console.log('Saavn.com connector loading');
+
+// Track enclosing <div>, id: 'now-playing' to observe track change
+$('#now-playing').on('DOMSubtreeModified', function () {
```
Comment (dufferzafar) [inline connectors/saavn.js]: The `.on` function is available in jquery > 1.7, but for v1 connectors we include jquery 1.6 

I got things to work by just using jquery 2 for this script.

But I think we should just port this connector to v2

@alexesprit Can you please have a look at tell me how can we achieve the ajax call etc. in v2 format?

File: connectors/saavn.js (hunk 1 above)
Comment (dufferzafar) [inline connectors/saavn.js]: I don't think gaana.com has links for a single track, like saavn does.

File: connectors/saavn.js (hunk 2 above)
Comment (alexesprit) [inline connectors/saavn.js]: @dufferzafar ok, I'll take a look at it. I have some thoughts about ajax in v2 connectors, but don't sure if it helps.

File: connectors/saavn.js (hunk 1 above)
Comment (alexesprit) [inline connectors/saavn.js]: gaana.com has link to the album page, that contains now playing song info, including artist name.

File: connectors/saavn.js (hunk 2 above)
Comment (alexesprit) [inline connectors/saavn.js]: > The .on function is available in jquery > 1.7, but for v1 connectors we include jquery 1.6

You could try to use MutationObserver to process DOM changes, though. That's how v2 core work (and some v1 connectors too).

But v2 format is preferred anyway.

File: connectors/saavn.js (hunk 1 above)
Comment (dufferzafar) [inline connectors/saavn.js]: There's a link to the album page, yes, which includes the information but there's no indicator as to which song is now playing, so we'll have to find that from the list, right?

Comment (alexesprit) [inline connectors/saavn.js]: image
The `ul` element of now playing song has `currentSong` class, by which we can identify now playing song.
I also tried to play several songs on the same page, this `currentSong` class is set to proper song on every song change. Some websites are buggy and don't switch off indicators like this one, so they have a lot of marked as 'now playing' songs.

Comment (dufferzafar) [inline connectors/saavn.js]: Okay, but this is being dynamically added, right? We'll have to see if that `currentSong` indicator is available to us in the HTML fetched via Ajax.

Comment (alexesprit) [inline connectors/saavn.js]: Agree, it's need to implement AJAX fetching, and we'll see how we can use that track info.

File: connectors/saavn.js (hunk 2 above)
Comment (inverse) [inline connectors/saavn.js]: V1 is definitely deprecated. I'll update the wiki

**Comment by alexesprit** [comment]
I didn't test the connector too much, but it should work without problems.
```js
'use strict';

/* global Connector */

var songArtist = null;
var trackInfoUrl = null;

var asyncRequestSongInfo = function() {
    var url = $('#player-track-name a').attr('onclick').split(',')[1].slice(2, -3);
    if (trackInfoUrl != url) {
        $.ajax({url: url, context: document.body}).done(function(trackDetailsHTML) {
            var trackDetailsDOM = $(trackDetailsHTML);
            var artist = trackDetailsDOM.find('.page-meta-group > .meta-list')[0].textContent;
            songArtist = artist;
        });
        trackInfoUrl = url;
    }    
};

Connector.playerSelector = '#now-playing';

Connector.getArtist = function() {
    asyncRequestSongInfo();
    return songArtist;
};

Connector.trackSelector = '#player-track-name';

Connector.isPlaying = function() {
    return $('#controls #play').hasClass('hide');
};
```

**Comment by alexesprit** [comment]
@dufferzafar I implemented the connector in v2 format some time ago. Can you test it?

**Comment by alexesprit** [comment]
We dropped support for v1 format.

**Pull request closed without merging**